src/backend/drm/compositor/mod.rs

```diff
@@ -1006,14 +1073,15 @@ where
         }
 
         // first do the potential blit
-        if let Some((dmabuf, geometry)) = primary_dmabuf {
+        if let Some((sync, dmabuf, geometry)) = primary_dmabuf {
             let blit_damage = damage
                 .iter()
                 .filter_map(|d| d.intersection(geometry))
                 .collect::<Vec<_>>();
 
             trace!("blitting frame with damage: {:#?}", blit_damage);
 
+            sync.wait();
```

Comment:
Sorry I am only seeing this now, but I am a little worried again, that this might cause unnecessary blocks in the client, but for the screencopy use-case.

It totally makes sense for the `commit` and `page_flip` functions, but here we have a renderer. Can't and shouldn't we do `renderer.wait(&sync)` here?

Reply:
yeah, I replaced it with a potential server wait and also changed the function to return a `SyncPoint` instead of waiting for the rendering of overlay elements.